sdk/storage/storage-file-share/src/ShareFileClient.ts

```diff
@@ -2475,4 +2622,261 @@ export class ShareFileClient extends StorageClient {
+  private _leaseId: string;
+  private _url: string;
+  private _file: File;
+  /**
+   * Gets the lease Id.
+   *
+   * @readonly
+   * @memberof ShareLeaseClient
+   * @type {string}
+   */
+  public get leaseId(): string {
+    return this._leaseId;
+  }
+
+  /**
+   * Gets the url.
+   *
+   * @readonly
+   * @memberof ShareLeaseClient
+   * @type {string}
+   */
+  public get url(): string {
+    return this._url;
+  }
+
+  /**
+   * Creates an instance of ShareLeaseClient.
+   * @param {ShareFileClient} client The client to make the lease operation requests.
+   * @param {string} leaseId Initial proposed lease id.
+   * @memberof ShareLeaseClient
+   */
+  constructor(client: ShareFileClient, leaseId?: string) {
+    const clientContext = new StorageClientContext(
+      SERVICE_VERSION,
+      client.url,
+      (client as any).pipeline.toServiceClientOptions()
+    );
+    this._file = new File(clientContext);
+
+    this._url = client.url;
+
+    if (!leaseId) {
+      leaseId = generateUuid();
+    }
+    this._leaseId = leaseId;
+  }
+
+  /**
+   * The Lease File operation establishes and manages a lock on a file for write and delete operations.
+   *
+   * @param {number} duration Specifies the duration of lease. The only allowed value is -1, for a lease that never expires.
+   * @param {LeaseOperationOptions} [options={}] Options for the lease management operation.
+   * @returns {Promise<LeaseOperationResponse>} Response data for acquire lease operation.
+   * @memberof ShareLeaseClient
+   */
+  public async acquireLease(
+    duration = -1,
+    options: LeaseOperationOptions = {}
+  ): Promise<LeaseOperationResponse> {
+    const { span, spanOptions } = createSpan(
+      "ShareLeaseClient-acquireLease",
+      options.tracingOptions
+    );
+    try {
+      return await this._file.acquireLease({
+        abortSignal: options.abortSignal,
+        duration,
+        proposedLeaseId: this._leaseId,
+        spanOptions
+      });
+    } catch (e) {
+      span.setStatus({
+        code: CanonicalCode.UNKNOWN,
+        message: e.message
+      });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * To change the ID of an existing lease.
+   *
+   * @param {string} proposedLeaseId the proposed new lease Id.
+   * @param {LeaseOperationOptions} [options={}] Options for the lease management operation.
+   * @returns {Promise<LeaseOperationResponse>} Response data for change lease operation.
+   * @memberof ShareLeaseClient
+   */
+  public async changeLease(
+    proposedLeaseId: string,
+    options: LeaseOperationOptions = {}
+  ): Promise<LeaseOperationResponse> {
+    const { span, spanOptions } = createSpan("ShareLeaseClient-changeLease", options.tracingOptions);
+    try {
+      const response = await this._file.changeLease(this._leaseId, {
+        proposedLeaseId,
+        abortSignal: options.abortSignal,
+        spanOptions
+      });
+      this._leaseId = proposedLeaseId;
```

Comment:
this._leaseId = proposedLeaseId; 

This one might be interesting. Think about some user invokes changeLease without await, and triggers some other operations about leasing, those operations behavior might be hard to predict.

Reply:
Let's live with it for now. We can further discuss about this.